File: backend/semgrep_rules/tests/javascript/kisa_sw_22_cleartext_sensitive_data.js
const axios = require("axios");
function unsafe(payload) {
  // ruleid: kisa.sw22.javascript.cleartext-http-transport-review
  return axios.post("http://api.example.com/login", payload);
}
function safe(payload) {
  // ok: kisa.sw22.javascript.cleartext-http-transport-review
  return axios.post("https://api.example.com/login", payload);
}

File: backend/semgrep_rules/tests/python/kisa_sw_22_cleartext_sensitive_data.py
import requests
def unsafe(payload):
    # ruleid: kisa.sw22.python.cleartext-http-transport-review
    return requests.post("http://api.example.com/login", data=payload)
def safe(payload):
    # ok: kisa.sw22.python.cleartext-http-transport-review
    return requests.post("https://api.example.com/login", data=payload)
